docs/src/DevelopersGuide.md

```diff
@@ -0,0 +1,142 @@
+    Any errors will be related to documentation issues such as missing docstrings.
+The ``CodeCov`` and ``ci / ci 1.8.1 `` checks look at CloudMicrophysics library
+    as a whole. Oftentimes, if your files run fine locally, issues are related to
+    tests. See the `Tests` section below for debugging tips.
+
+Before merging, you can comment `bors try` on your PR to trigger a `build kite` check.
+    This will check if your code can run properly on GPU. If this check fails despite all
+    other checks succeeding, it is likely because a function used is not able to run on GPU.
+    Check your GPU tests for use of any out of place functions/commands (for example, `@warn`
+    will not work on GPU).
+
+Once someone has reviewed your PR and approved it,
+    make sure to squash all commits so that the PR only has 1 commit.
+    To merge to main, comment `bors r+` and wait for all checks to pass.
+
+## New Functions
+
+When creating a new function, remember to include docstrings and tests.
+    Docstrings should be written immediately before a new function or module,
+    no new lines in between. It should start and end with a line consisting of
+    three consecutive quotation marks. The second line should have the function's
+    name preceeded by exactly 4 spaces so that Julia can recognize it as a
+    docstring. Examples can be found in the source files immediately before
+    any function or module is defined. Be sure to also add the function to `API.md`
+    in the docs folder so that it can be properly included in the building of the
+    documentation.
+
+All functions should have their own performance, GPU, and unit
+    tests. See the ``Tests`` section below for more information.
+
+Other files that may require editing after you make a new function are
+ - `index.md` (found in docs folder) under the list of things that CloudMicrophysics.jl includes
+ - `make.jl` (found in docs folder) if you are adding a new file
+ - `runtests.jl` (found in test folder) if you are adding new test files for unit tests
+
+## Tests
+
+### Performance Tests
+
+Performance tests check the performance of functions.
+    They are found in the test folder under a single file named `performance_tests.jl`.
+
+You can add performance tests for your new functions in the `benchmark_test(FT)`
+    function. Parameters are listed at the start of the function. If there is a
+    parameter that your function needs which is not listed inside the function, add them
+    here. Next, add your performance tests under the comment with the file name which your
+    new function is in. This is done by calling `bench_press(function_name, (Parameter1, Parameter2), 40)`.
+    The last argument is an estimate of the minimum runtime of the function. At first, you
+    can use 40. Run the file to see if the tests run successfully. If a test fails, the terminal
+    will output the actual minimum runtime. Replace 40 with double of what the terminal
+    states the runtime is. 
```

Comment:
```suggestion
    The last argument is an estimate of the minimum runtime of the function. It may take some trial and error
    to find a number that will satisfy the `ci` tests that are run on different operating systems.
```